### DISCUSSION TITLE
User access to project reports

### DISCUSSION THREAD

**Discussion by damsleth** [Puzzlepart/did#969] [Feature request]
## What
`Users` don't have access to reports. 
This means they can't see the `Reports` tab (which is fine),
but they also can't see time entries for a project, since it includes hours from other users.

(See [related slack thread](https://puzzlepart.slack.com/archives/GQK65AT0S/p1621508929018200))

There are several ways to solve this issue, but we have to figure out which solution to choose.

## Some solution alternatives
1. Hide the `Time entries` button from the `ProjectDetails` panel for `Users`
2. Give `Users` access to the `ProjectReport` operation (but not `SummaryReport`)
3. Let `Users` access a subset of `ProjectReport`, with just their own time entries

**Comment by olemp** [comment]
A fourth alternative is to have project scoped permissions for the project reports. E.g. project manager can retrieve the hours.

**Reply by damsleth** [reply]
Allright, how to implement?
A `Project manager` user-field under project details for instance?
![image](https://user-images.githubusercontent.com/7300548/118973892-78db9980-b972-11eb-9ce7-94b18b195408.png)

**Reply by olemp** [reply]
Yes, or a multi user field with the users that should have access to retrieve the hours.

**Reply by olemp** [reply]
So either we restrict access on project level using some user fields, or we have separate permissions for project reports.

**Reply by paalolav** [reply]
I believe a summary of the user's hours on the projects the user is involved in would be beneficial for everyone.
A project-manager would need to be able to retrieve the hours, so a scoped permission for the project reports would also be great.

In other words suggestion 1/3 and 4 got my vote. (Written in fluid norwenglish)

**Reply by pettersh** [reply]
I'need this: 

- "As a delivery/project manager I to report hours on my delivery/project"
- "As an account manager I need to report hours on my customer

I would most likely need different reports

- weekly/monthly/year to date
- summary of hours/per project/per user/time entry details

**Comment by olemp** [comment]
Handled for now by adjusting the User role:

![image](https://user-images.githubusercontent.com/7606007/221580277-639c0399-fdb5-44d7-91fd-9e5e851a3d6e.png)
